pysteps/nowcasts/linear_blending.py

```diff
@@ -0,0 +1,172 @@
+      Name of the nowcast method used to forecast the precipitation.
+    R_nwp: array_like or NoneType, optional
+      Array of shape (timesteps, m, n) in the case of no ensemble or
+      of shape (n_ens_members, timesteps, m, n) in the case of an ensemble
+      containing the NWP precipitation fields ordered by timestamp from oldest
+      to newest. The time steps between the inputs are assumed to be regular
+      (and identical to the time step between the nowcasts). If no NWP
+      data is given the value of R_nwp is None and no blending will be performed.
+    start_blending: int, optional
+      Time stamp (in minutes) after which the blending should start. Before this
+      only the nowcast data is used.
+    end_blending: int, optional
+      Time stamp (in minutes) after which the blending should end. Between
+      start_blending and end_blending the nowcasts and NWP data are linearly
+      merged with each other. After end_blending only the NWP data is used.
+    nowcast_kwargs: dict, optional
+      Dictionary containing keyword arguments for the nowcast method.
+
+    Returns
+    -------
+    R_blended: ndarray
+      Array of shape (timesteps, m, n) in the case of no ensemble or
+      of shape (n_ens_members, timesteps, m, n) in the case of an ensemble
+      containing the precipation forecast generated by linearly blending
+      the nowcasts and the NWP data.
+    """
+
+    # Calculate the nowcasts
+    nowcast_method_func = nowcasts.get_method(nowcast_method)
+    R_nowcast = nowcast_method_func(
+        precip,
+        velocity,
+        timesteps,
+        **nowcast_kwargs,
+    )
+
+    # Transform the precipitation back to mm/h
+    R_nowcast = transformation.dB_transform(R_nowcast, threshold=-10.0, inverse=True)[0]
+
+    # Check if NWP data is given as input
+    if R_nwp is not None:
+
+        if len(R_nowcast.shape) == 4:
+            n_ens_members_nowcast = R_nowcast.shape[0]
+            if n_ens_members_nowcast == 1:
+                R_nowcast = np.squeeze(R_nowcast)
+        else:
+            n_ens_members_nowcast = 1
+
+        if len(R_nwp.shape) == 4:
+            n_ens_members_nwp = R_nwp.shape[0]
+            if n_ens_members_nwp == 1:
+                R_nwp = np.squeeze(R_nwp)
+        else:
+            n_ens_members_nwp = 1
+
+        n_ens_members_max = max(n_ens_members_nowcast, n_ens_members_nwp)
+        n_ens_members_min = min(n_ens_members_nowcast, n_ens_members_nwp)
+
+        if n_ens_members_min != n_ens_members_max:
+            if n_ens_members_nwp == 1:
+                R_nwp = np.repeat(R_nwp[np.newaxis, :, :], n_ens_members_max, axis=0)
+            elif n_ens_members_nowcast == 1:
+                R_nowcast = np.repeat(
+                    R_nowcast[np.newaxis, :, :], n_ens_members_max, axis=0
+                )
+            else:
+                repeats = [
```

Comment:
After consulting @wdewettin and @ladc, I figured out what this piece of code does. So if you e.g. have 10 ensemble nowcasts members and 3 NWP members, the output will be an ensemble of 10 members. Hence, the three NWP members are blended with the first three members of the nowcast (member one with member one, two with two, etc.), subsequently, the same NWP members are blended with the next three members (NWP member one with member 4, NWP member 2 with member 5, etc.), until 10 is reached.

Is this the way to go (works for me) or do we want an output of 30 (10 x 3 members here) in this case?

Reply:
Oh and @wdewettin, let's add some comments to the code here so that this procedure is clear. :) (B.t.w., I can add that too, but I'll wait for the responses - and thus possible changes - of the others)

Reply:
Yes! I will await the responses from the others, and then add some comments